worlds/terraria/__init__.py

```diff
@@ -0,0 +1,311 @@
+                if type(arg) is not int:
+                    raise Exception("@pickaxe requires an integer argument")
+
+                for pickaxe, power in pickaxes.items():
+                    if power >= arg and state.has(pickaxe, self.player):
+                        return sign
+
+                return not sign
+            elif condition == "hammer":
+                if type(arg) is not int:
+                    raise Exception("@hammer requires an integer argument")
+
+                for hammer, power in hammers.items():
+                    if power >= arg and state.has(hammer, self.player):
+                        return sign
+
+                return not sign
+            elif condition == "mech_boss":
+                if type(arg) is not int:
+                    raise Exception("@mech_boss requires an integer argument")
+                boss_count = 0
+                for boss in mech_bosses:
+                    if state.has(boss, self.player):
+                        boss_count += 1
+                        if boss_count >= arg:
+                            return sign
+
+                return not sign
+            else:
+                raise Exception(f"Unknown function {condition}")
+        elif ty == COND_GROUP:
+            operator, conditions = condition
+            return sign == self.check_conditions(state, operator, conditions)
+
+    def check_conditions(
+        self,
+        state,
+        operator: bool | None,
+        conditions: list[
+            tuple[bool, int, str | tuple[bool | None, list], str | int | None]
+        ],
+    ) -> bool:
+        if operator is None:
+            if len(conditions) == 0:
+                return True
+            if len(conditions) > 1:
+                raise Exception("Found multiple conditions without an operator")
+            return self.check_condition(state, *conditions[0])
+        elif operator:
+            return any(
+                self.check_condition(state, *condition) for condition in conditions
+            )
+        else:
+            return all(
+                self.check_condition(state, *condition) for condition in conditions
+            )
+
+    def set_rules(self) -> None:
+        for location in self.ter_locations:
+
+            def check(state: CollectionState, location=location):
+                _, _, operator, conditions = rules[rule_indices[location]]
+                return self.check_conditions(state, operator, conditions)
+
+            self.multiworld.get_location(location, self.player).access_rule = check
+
+    def generate_basic(self) -> None:
+        for location in self.ter_locations:
+            _, flags, _, _ = rules[rule_indices[location]]
+            if not "Location" in flags and not "Achievement" in flags:
+                if location in progression:
+                    classification = ItemClassification.progression
+                else:
+                    classification = ItemClassification.useful
```

Comment:
Item creation ideally should be done in `def create_items(self) -> None:` as per #1460